src/main/java/com/ibm/watson/modelmesh/Metrics.java

```diff
@@ -230,6 +229,33 @@ public PrometheusMetrics(Map<String, String> params) throws Exception {
                 }
             }
 
+            if (!infoMetricParams.isEmpty()){
+                if (infoMetricParams.size() > INFO_METRICS_MAX) {
+                    throw new Exception("Too many info metrics provided in env var " + MMESH_CUSTOM_ENV_VAR + ": \""
+                            + infoMetricParams+ "\". The max is " + INFO_METRICS_MAX);
+                }
+
+                // Remove unset (unresolved) labels
+                for(String key : infoMetricParams.keySet()) {
+                    if(infoMetricParams.get(key).contains("$(")){
+                        infoMetricParams.remove(key);
+                        logger.info(("Info metric for label '" + key + "' is missing in env var: "
+                                + MMESH_CUSTOM_ENV_VAR));
+                    }
+                }
+
+                String metric_name = infoMetricParams.remove("metric_name");
+                String[] labelNames = infoMetricParams.keySet().toArray(new String[0]);
+                String[] labelValues = infoMetricParams.values().toArray(new String[0]);
```

Comment:
I don't think it's a great idea to rely on the fact that the passed in hashmap is ordered. For a regular hashmap there's no guarantee these names/values will end up in the same order. Better to create two empty arrays of the correct size prior to the loop above and just populate the names/values in the loop.

Reply:
I was using a linkedhashmap here to preserve the order, I made it a bit more obvious now. Interested to know what you think or if the array creation method would still be better.

Reply:
Yep I did see the LinkedHashMap but I still think it's cleaner not require it as the type. You could do:
```java
String[] labelNames = infoMetricParams.keySet().toArray(String[]::new);
String[] labelValues = Stream.of(labelNames).map(infoMetricParams::get).toArray(String[]::new);
```